File: 2497-maximum-matching-of-players-with-trainers/maximum-matching-of-players-with-trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort()
        trainers.sort()
        i = 0
        res = 0
        for j in range(len(trainers)):
            if players[i] <= trainers[j]:
                res += 1
                i += 1
                if i == len(players):
                    break
        return res

File: 2497-maximum-matching-of-players-with-trainers/maximum-matching-of-players-with-trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        sort(players.begin(),players.end());
        sort(trainers.begin(),trainers.end());
        int res = 0;
        int i = 0;
        int p = players.size();
        int t = trainers.size();
        for(int j = 0 ; j < t ; j++){
            if(players[i] <= trainers[j]){
                res++;
                i++;
                if(i == p)
                    break;
            }
        }
        return res;
    }
};